DataGrid: add Disable Editing section (#4455)
DevExpress/devextreme-documentation: concepts/05 UI Components/DataGrid/20 Editing/45 Disable Editing/30 Row.md

> If you want to forbid editing in a specific row, take the following steps:

Reviewer: When you give step-by-step instructions, it's better to use an ordered list.

Author: ok

> If you want to forbid editing in a specific row, take the following steps:

Reviewer 2: "Forbid" is more like an order not to do something, but it kind of implies that I still can do it. (I can forbid my son to play in the rain... but it doesn't mean I took away that possibility. I trust he will follow my command even though the opportunity is present.) 

Anyway I wouldn't use this word in a technical context. 

```suggestion
If you want to disable data editing within a specific row, take the following steps:
```

> If you want to disable data editing within a specific row, take the following steps:

Reviewer 3: ```suggestion
To disable data editing within a specific row, follow the steps below:
```

> 1. Handle the DataGrid [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event. 

Reviewer 2: ```suggestion
1. Handle the DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event. 
```

> 2. Check the **data** parameter's values and assign **true** to **e.cancel**.

Reviewer 2: I would add a bit more details about what "true" means. 

```suggestion
2. Check the **data** parameter's values. If you want don't want users to edit this row, assign **true** to **e.cancel**.
```

        }
    }
    @NgModule({
        imports: [
            // ...
            DxDataGridModule
        ],
        // ...
    })

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxDataGrid 
            @editing-start="onEditingStart"
        >
        </DxDataGrid>
    </template>

    <script>
    import {
        DxDataGrid,
        // ...
    } from 'devextreme-vue/data-grid';

    export default {
        components: {
            DxDataGrid,
            // ...
        },
        methods: {
            onEditingStart(e) {
                e.cancel = e.data.Position === "CEO";
            }
        }
        // ...
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React, { useCallback } from "react";
    import 'devextreme/dist/css/dx.light.css';

    import { DataGrid } from 'devextreme-react/data-grid';

    function App() {
        // ...
        const onEditingStart = useCallback((e) => {
           e.cancel = e.data.Position === "CEO";
        }, []);

        return (
            <DataGrid 
                onEditingStart={onEditingStart}
            >
            </DataGrid>
        );
    }

    export default App;

---

> The DataGrid [editing mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#mode) defines whether to display the **Edit** button opposite every row. To hide this button for the required rows, use the button's [visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/buttons/#visible) property. You can also assign a function to this property and check a row's data in a similar manner as in the DataGrid [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) case.

Reviewer 2: "you can also assign a function" - I wouldn't say that. In this case, we are already describing a specific scenario. I want to disable the button for specific rows. So I can proceed to direct instructions. "Assign a function..."

```suggestion
The DataGrid's [editing mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#mode) defines whether to display the **Edit** button opposite every row. To hide this button for the required rows, use the button's [visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/buttons/#visible) property. Assign a function to this property and check a row's data in a similar manner as in the DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event.
```

> The DataGrid's [editing mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#mode) defines whether to display the **Edit** button opposite every row. To hide this button for the required rows, use the button's [visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/buttons/#visible) property. Assign a function to this property and check a row's data in a similar manner as in the DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event.

Reviewer 3: what do you mean by required row?
when you say assign a function, what function to do you mean?
"similar manner" - can you specify?

Author: **what do you mean by required row?** - It is the rows, for which you want to hide the Edit buttons

**when you say assign a function, what function to do you mean?** - this function with condition is described in the code snippets below: 

visible: function(e) {
      return e.row.data.Position !== "CEO";
}

**"similar manner" - can you specify?** - there is the link to code snippets, where users can see the "similar manner", if the code snippets below are not clear for them

Reviewer 3: ```suggestion
The DataGrid's [editing mode](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#mode) defines whether to display the **Edit** button opposite every row. To hide this button, use the button's [visible](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/buttons/#visible) property. Assign the [PLEASE PROVIDE NAME OF FUNCTION] function to this property and ...
```
"check row data in a similar manner as in the DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event.
When you say "check", do you mean review? Are you trying to say review DataGrid's [onEditingStart](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/#onEditingStart) event for similar functionality?

Author: Ok! There is the **visible** property.  You can set it to _boolean_ value or assign the callback function to this property. In our case, the callback function is anonymous, so I cannot provide its name.

It is the example:

visible: function(e) { _// it is anonymous function, that returns a boolean value!_
     return e.row.data.Position !== "CEO";  _// it is the row data the user need to check!_
}

Text can be as follow: The DataGrid's editing mode defines whether to display the Edit button opposite every row. To hide this button, use the button's visible property. Assign a callback to this property and check row's data within this function.

Reviewer 3: DataGrid's editing mode defines whether to display an Edit button opposite every row. To hide this button, use the button's visible property. Assign a callback to this property and check row data within this function.
